ktor-server/ktor-server-core/jvm/src/io/ktor/routing/RoutingBuilder.kt

```diff
@@ -118,11 +118,15 @@ public fun Route.post(path: String, body: PipelineInterceptor<Unit, ApplicationC
 @ContextDsl
 @JvmName("postTyped")
 public inline fun <reified R : Any> Route.post(
-    path: String,
+    path: String? = null,
     crossinline body: suspend PipelineContext<Unit, ApplicationCall>.(R) -> Unit
 ): Route {
-    return route(path, HttpMethod.Post) {
-        handle {
+    return if(path != null) {
```

Comment:
The problem here is that the function is inline so introducing `if` here you duplicate the body bytecode and duplicate user's bytecode everywhere regardless of whether they need the `path` or not.

Reply:
So you need multiple overloads? One with path, one without?